synapse/http/matrixfederationclient.py

```diff
@@ -212,6 +213,11 @@ def _send_request(
 
             backoff_on_404 (bool): Back off if we get a 404
 
+            try_trailing_slash_on_404 (bool): True if on a 404 response we
```

Comment:
it might be worth adding a note on why this is a useful thing to do ("workaround for #3622 in Synapse 0.99.2 and earlier" ?)